Source/Blazorise/Components/Modal/Modal.razor.cs

```diff
@@ -313,26 +313,53 @@ protected virtual void HandleVisibilityStyles( bool visible )
                 } );
             }
 
-            DirtyClasses();
-            DirtyStyles();
+            await ModalDirtyStylingAnimation( visible );
+        }
+
+        /// <summary>
+        /// Handles the modal styling in a way that still triggers css animations.
+        /// </summary>
+        /// <param name="visible"></param>
+        /// <returns></returns>
+        private async Task ModalDirtyStylingAnimation( bool visible )
+        {
+            var hasFadeAnimation = !string.IsNullOrWhiteSpace( ClassProvider.ModalFade() );
+
+            if ( visible )
+                DirtyStyles();
+            else
+                DirtyClasses();
+
+            if ( hasFadeAnimation )
+            {
+                await InvokeAsync( StateHasChanged );
+                await Task.Delay( 150 );
```

Comment:
Maybe we can add animation duration as a parameter?

Reply:
Sure. Not sure how much boiler plate needed for that yet.